### ISSUE TITLE
six text_types or text_type

### ISSUE THREAD

**Issue by matsepp** [yhat/ggplot#505]
Hi, I just tried ggplot for the first time and got this:

ggplot.py", line 151, in add_labels
    if isinstance(label, six.text_types):
AttributeError: module 'six' has no attribute 'text_types'

with six-1.10.0, should be the last one.

I had to change this line  six.text_types to  six.text_type, is it a know bug ?

Mathieu

**Comment by suensummit** [comment]
The same error here, even running the example codes on the official site.
```{python}
from ggplot import *

ggplot(aes(x='date', y='beef'), data=meat) +\
    geom_line() +\
    stat_smooth(colour='blue', span=0.2)
```
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/home/summit/.pyenv/versions/3.5.1/lib/python3.5/site-packages/IPython/core/formatters.py in __call__(self, obj)
    697                 type_pprinters=self.type_printers,
    698                 deferred_pprinters=self.deferred_printers)
--> 699             printer.pretty(obj)
    700             printer.flush()
    701             return stream.getvalue()

/home/summit/.pyenv/versions/3.5.1/lib/python3.5/site-packages/IPython/lib/pretty.py in pretty(self, obj)
    381                             if callable(meth):
    382                                 return meth(obj, self, cycle)
--> 383             return _default_pprint(obj, self, cycle)
    384         finally:
    385             self.end_group()

/home/summit/.pyenv/versions/3.5.1/lib/python3.5/site-packages/IPython/lib/pretty.py in _default_pprint(obj, p, cycle)
    501     if _safe_getattr(klass, '__repr__', None) not in _baseclass_reprs:
    502         # A user-provided repr. Find newlines and replace them with p.break_()
--> 503         _repr_pprint(obj, p, cycle)
    504         return
    505     p.begin_group(1, '<')

/home/summit/.pyenv/versions/3.5.1/lib/python3.5/site-packages/IPython/lib/pretty.py in _repr_pprint(obj, p, cycle)
    692     """A pprint that just redirects to the normal repr function."""
    693     # Find newlines and replace them with p.break_()
--> 694     output = repr(obj)
    695     for idx,output_line in enumerate(output.splitlines()):
    696         if idx:

/home/summit/.pyenv/versions/3.5.1/lib/python3.5/site-packages/ggplot/ggplot.py in __repr__(self)
    114 
    115     def __repr__(self):
--> 116         self.make()
    117         # this is nice for dev but not the best for "real"
    118         if os.environ.get("GGPLOT_DEV"):

/home/summit/.pyenv/versions/3.5.1/lib/python3.5/site-packages/ggplot/ggplot.py in make(self)
    603             self.add_labels()
    604             self.apply_axis_scales()
--> 605             self.apply_axis_labels()
    606             self.apply_coords()
    607             self.add_legend(legend)

/home/summit/.pyenv/versions/3.5.1/lib/python3.5/site-packages/ggplot/ggplot.py in apply_axis_labels(self)
    270 
    271         if xlab:
--> 272             if isinstance(xlab, six.text_types):
    273                 xlab = element_text(xlab)
    274 

AttributeError: module 'six' has no attribute 'text_types'
```

**Comment by williford** [comment]
@nicktimko can you fix this? "text_types" needs to be "text_type".

**Comment by schmeing** [comment]
I tested simply changing it, but got:

```python
$ python
Python 2.7.6 (default, Jun 22 2015, 17:58:13) 
[GCC 4.8.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from ggplot import *
>>> p = ggplot(aes(x='mpg'), data=mtcars) + geom_histogram()
>>> print p
/usr/local/lib/python2.7/dist-packages/ggplot/ggplot.py:203: RuntimeWarning: Setting "mpl.rcParams['axes.prop_cycle']=cycler(u'color', [u'#333333', u'#348ABD', u'#7A68A6', u'#A60628', u'#467821', u'#CF4457', u'#188487', u'#E24A33'])" raised an Exception: u'axes.prop_cycle is not a valid rc parameter.See rcParams.keys() for a list of valid parameters.'
  warnings.warn(msg, RuntimeWarning)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/dist-packages/ggplot/ggplot.py", line 116, in __repr__
    self.make()
  File "/usr/local/lib/python2.7/dist-packages/ggplot/ggplot.py", line 605, in make
    self.apply_axis_labels()
  File "/usr/local/lib/python2.7/dist-packages/ggplot/ggplot.py", line 276, in apply_axis_labels
    xlab.override(0.5, 0.05)
AttributeError: 'str' object has no attribute 'override'
>>> p.save("myplot.png")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/dist-packages/ggplot/ggplot.py", line 552, in save
    self.make()
  File "/usr/local/lib/python2.7/dist-packages/ggplot/ggplot.py", line 605, in make
    self.apply_axis_labels()
  File "/usr/local/lib/python2.7/dist-packages/ggplot/ggplot.py", line 276, in apply_axis_labels
    xlab.override(0.5, 0.05)
AttributeError: 'str' object has no attribute 'override'
```

**Comment by glamp** [comment]
fixed in 0e05ea562ac30071f123a11da4b18e80e307fc15

**Comment by glamp** [comment]
Getting a 500 server error when uploading to PyPi. Will reopen until this is live...

```
...
creating build/bdist.macosx-10.6-x86_64/wheel/ggplot-0.10.3.dist-info/WHEEL
running upload
Submitting dist/ggplot-0.10.3-py2.7.egg to https://pypi.python.org/pypi
Upload failed (500): Internal Server Error
error: Upload failed (500): Internal Server Error
```

**Comment by glamp** [comment]
ok officially fixed in 0.10.4!
